firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayView.swift

```diff
@@ -102,12 +102,24 @@ class TabDisplayView: UIView,
     }
 
     func newState(state: TabsPanelState) {
+        // We only want to respond to `TabsPanelState` for the current `panelType` (e.g. don't scroll the regular tabs for
+        // a private tabs scroll)
+        switch panelType {
+        case .tabs:
+            guard !state.isPrivateMode else { return }
+        case .privateTabs:
+            guard state.isPrivateMode else { return }
+        case .syncedTabs:
+            // FIXME what about the synced tab panel?
+            break
+        }
+
```

Comment:
@dataports Between this added logic and line 138 below (`let section: Int = scrollState.isInactiveTabSection ? 1 : 0`) we should be in a good spot for not getting any crashes related to scrolling to out of bounds indices. (Can't promise this will fix your batch update problem, but hopefully it helps!)

I'm guessing at big part of the issue was that we weren't filtering the `TabsPanelState` for the correct panel type.